src/custom_assets.rs

```diff
@@ -268,8 +360,13 @@ pub fn custom_assets<'r>(
 pub fn collect_game_resources<'r>(
     gc_disc: &structs::GcDisc<'r>,
     starting_memo: Option<&str>,
+    config: &PatchConfig,
 )
-    -> HashMap<(u32, FourCC), structs::Resource<'r>>
+    -> (
+        HashMap<(u32, FourCC), structs::Resource<'r>>,
+        HashMap<PickupHashKey, ResId<res_id::STRG>>,
+        HashMap<PickupHashKey, (ResId<res_id::SCAN>, ResId<res_id::STRG>)>,
+    )
```

Comment:
This return type is kind of confusing. You have to look at the body of the function to work out what the latter two hash maps are supposed to be used for. Can you add a `TODO` note indicating it should be cleaned up?